migrator/zenodo_rdm_migrator/actions/users.py

```diff
@@ -0,0 +1,152 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Migrator is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Invenio RDM migration users actions module."""
+
+
+from invenio_rdm_migrator.actions import TransformAction
+from invenio_rdm_migrator.load.postgresql.transactions.operations import OperationType
+from invenio_rdm_migrator.streams.actions import (
+    UserDeactivationAction,
+    UserEditAction,
+    UserRegistrationAction,
+)
+from invenio_rdm_migrator.transform import IdentityTransform
+
+from ..transform.entries.users import ZenodoUserEntry
+
+
+class ZenodoUserRegistrationAction(TransformAction):
+    """Zenodo to RDM user registration action."""
+
+    name = "register-user"
+    load_cls = UserRegistrationAction
+
+    @classmethod
+    def matches_action(cls, tx):  # pragma: no cover
+        """Checks if the data corresponds with that required by the action."""
+        rules = {
+            "userprofiles_userprofile": OperationType.INSERT,
+            "accounts_user": OperationType.INSERT,
+        }
+
+        for operation in tx.operations:
+            table_name = operation["source"]["table"]
+
+            rule = rules.pop(table_name, None)
+            # no duplicated tables, can fail fast if rule is None
+            if not rule or not rule == operation["op"]:
+                return False
+
+        return len(rules) == 0
+
+    def _transform_data(self):  # pragma: no cover
+        """Transforms the data and returns an instance of the mapped_cls."""
+        payload = {}
+        for operation in self.tx.operations:
+            payload = {**payload, **operation["after"]}
```

Comment:
minor: is this to merge all the entities into one dictionary? Aren't we afraid of overwriting/merging common keys (e.g. `updated` or `id`?).

I see that this will be a common thing, so maybe we could create some utilities on the `Tx`. class to make it easier to retrieve entries from a specific table from the `.operations` key.

Reply:
This one is done on purpose and it's meant to overwrited. The thing is this action gets multiple changes on the same row of the same table. Therefore, we only want the last one. At the moment, we could simplye do `operations[-1]` but that would be wrong if we have partial updates. With partial update the current code would end up with the sum of all changes and latest data. It's a bit "implicit". I can extract the behaviour to another function or leave a comment, however, I do not think is generic enough to move it to `Tx`. WDYT?

Reply:
I don't mean moving the ops-dict merging to `Tx`, but being able to retrieve all the operations from a specific table inside a `Tx`. E.g. in this case I'm worried that if there are more tables/models modified in the operations, you end up with invalid keys for the model. 

Although I see that it's passed to `ZenodoUserEntry.transform` later on, which is only picking specific fields. Anyways, not critical now, just a bit spooky to look at :)

Reply:
I understand now thanks, I think it could be helpful. I have a similar case now with the files update action. On the other hand, I _think_ it is not possible to end up with invalid keys since if there was a different table the `match_action` would not return `True` thus we would not go into tranforming the data. But I agree this is an "implicit truth" depending on a check.

I'll open an issue, we can discuss if it is priority or not.